controller/api/destination/watcher/cluster_store.go

```diff
@@ -247,21 +267,21 @@ func decodeK8sConfigFromSecret(data []byte, enableEndpointSlices bool) (*k8s.API
 			ctx,
 			cfg,
 			true,
-			k8s.ES, k8s.Pod, k8s.Svc, k8s.SP, k8s.Job, k8s.Srv,
```

Comment:
@mateiidavid I couldn't find where we used ServiceProfile or Node data so I removed them.  Do you know if these are actually used anywhere?

Reply:
The SP informer is still gonna get called via `NewServer()->watcher.NewProfileWatcher()` and if not initialized it will generate a panic? As for Nodes, we still use the endpoint translator, no?
Another thing I don't understand is why we need Job (here and in the original destination API) (introduced in 03762cc526723200db7f00204304e4ee410fc8a9).

Reply:
I was looking into the `Job` remark and it does seem like it's not used in the _endpoints watcher_. We get the owner kind and name here:

https://github.com/linkerd/linkerd2/blob/2d3f3d385c9f716dc0f867110c78c7a9e7eb0b8b/controller/api/destination/watcher/endpoints_watcher.go#L974-L977

Which in turn will use the metadata API to directly query the API server for Job/RS info. So while we need RBAC for `Jobs` it seems not to be required as an informer. It would seem better for the metadata client to also look for these resources in its cache. I think those are out of scope here tho.

Reply:
Thanks Matei. So we can get rid of `k8s.Job` when instantiating the `remoteAPI` but leave it in the instantiation of `metadataAPI`.

> It would seem better for the metadata client to also look for these resources in its cache

Don't we do this already? We call `api.getByNamespace()` and if that errors out, we fallback to the direct call.

About SP, you're right! (and now I get Node is only required in the metadataAPI, so that's fine too).